packages/liveblocks-core/src/protocol/UserNotificationSettings.ts

```diff
@@ -26,11 +27,52 @@ export type NotificationChannelSettings = {
 };
 
 /**
- * User notification settings are a set of notification channel setting.
+ * @private
+ *
+ * Base definition of user notification settings.
+ * Plain means it's a simple object coming from the remote backend.
+ *
+ * It's the raw settings object where somme channels cannot exists
+ * because there are no notification kinds enabled on the dashboard.
+ * And this object isn't yet proxied by the creator factory `createUserNotificationSettings`.
+ */
+export type UserNotificationSettingsPlain = {
+  [C in NotificationChannel]?: NotificationChannelSettings;
+};
+
+/**
+ * @private
+ *
+ * Symbol to brand some properties and methods
+ * as internal and private in `UserNotificationSettings`
+ */
+const kPrivate = Symbol("user-notification-settings-private");
+
+/**
+ * @private
+ *
+ * Private properties and methods internal to `UserNotificationSettings`.
+ * As a user of Liveblocks, you should NEVER USE ANY OF THESE DIRECTLY,
+ * because bad things will happen.
+ */
+export type PrivateUserNotificationSettingsApi = {
```

Comment:
This type doesn't seem to be used. Can you do a pass over all exported types that aren't used outside this module? Makes it much easier to verify we don't accidentally export private impl details.